Source/Libraries/GSF.Core/Diagnostics/ProcessUtilizationCalculator.cs

```diff
@@ -22,166 +22,256 @@
-                AutoReset = true,
-                Enabled = false
-            };
+        m_updateUtilizationTimer = new Timer(m_updateInterval)
+        {
+            AutoReset = true,
+            Enabled = false
+        };
 
-            m_updateUtilizationTimer.Elapsed += UpdateUtilizationTimerElapsed;
-        }
+        m_updateUtilizationTimer.Elapsed += UpdateUtilizationTimerElapsed;
+    }
 
-        #endregion
+    #endregion
 
-        #region [ Properties ]
+    #region [ Properties ]
 
-        /// <summary>
-        /// Gets the current processor utilization of the associated <see cref="Process"/>.
-        /// </summary>
-        public double Utilization { get; private set; }
+    /// <summary>
+    /// Gets the current processor utilization, percent between 0.0 and 1.0, of the associated <see cref="Process"/> set.
+    /// </summary>
+    public double Utilization { get; private set; }
 
-        /// <summary>
-        /// Gets or sets the interval, in milliseconds, over which <see cref="Utilization"/> will be calculated.
-        /// </summary>
-        public int UpdateInterval
+    /// <summary>
+    /// Gets or sets the interval, in milliseconds, over which <see cref="Utilization"/> will be calculated.
+    /// </summary>
+    public int UpdateInterval
+    {
+        get => m_updateInterval;
+        set
         {
-            get
-            {
-                return m_updateInterval;
-            }
-            set
-            {
-                m_updateInterval = value;
+            m_updateInterval = value;
 
-                if (m_updateInterval > 0)
-                    m_updateUtilizationTimer.Interval = m_updateInterval;
-            }
+            if (m_updateInterval > 0)
+                m_updateUtilizationTimer.Interval = m_updateInterval;
         }
+    }
 
-        /// <summary>
-        /// Gets associated process for this <see cref="ProcessUtilizationCalculator"/>.
-        /// </summary>
-        /// <remarks>
-        /// The <see cref="ProcessUtilizationCalculator"/> maintains a <see cref="WeakReference{T}"/> to the associated
-        /// <see cref="Process"/> so this property can be <c>null</c> if the process is no longer available.
-        /// </remarks>
-        public Process AssociatedProcess
-        {
-            get
-            {
-                Process process = null;
-                m_processReference?.TryGetTarget(out process);
-                return process;
-            }
-        }
+    /// <summary>
+    /// Gets associated processes for this <see cref="ProcessUtilizationCalculator"/>.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="ProcessUtilizationCalculator"/> maintains a <see cref="WeakReference{T}"/> to the associated
+    /// <see cref="Process"/> so this property can be <c>null</c> if the process is no longer available.
+    /// </remarks>
+    public Process[] AssociatedProcesses => m_processReferences.Select(reference => reference.Process).ToArray();
```

Comment:
Perhaps we should we filter out null here? I think the remarks need some tweaking either way.

```suggestion
    public Process[] AssociatedProcesses => m_processReferences.Select(reference => reference.Process).Where(process => process is not null).ToArray();
```

Reply:
I considered that. Since original property allowed a `null` return, thought I would leave the behavior with the only intelligence gain being "how many" child processes there possibly were - which may not be that useful. Although property is not used anywhere else (in library) and it's basically a breaking, so will apply suggestion - likely more useful that way anyway.